### ISSUE TITLE
Move Azure-specific documentation into an azure subdirectory

### ISSUE THREAD

**Issue by dthaler** [openenclave/openenclave#2759]
In PR#2659 Paul writes in comment https://github.com/openenclave/openenclave/pull/2659#discussion_r399520173 :
"lets leave this in the current directory and create a new issue about the possibility of splitting platform specific documentation out to platform specific directories, such as Azure, On-prem, and any other in the future"

Moving Azure-specific docs into a subdirectory makes it easier to separate OE core documentation from cloud-provider-specific information, and makes it easier to add non-Azure docs in parallel in an organized fashion.

**Comment by dthaler** [comment]
Paul also suggested there may be ways to collapse some information by default as discussed at https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab

**Comment by dthaler** [comment]
Another approach (TBD whether a good or bad idea) would be to allow markdown files to #include other markdown files, and generate the resulting built-out markdowns at build time using something like https://github.com/sethen/markdown-include#how-to-use-from-the-command-line

Then there could be a separate page for each combination, if we decided we wanted to go that route.

**Comment by AevaOnline** [comment]
I think, with a little effort, the docs in this folder could be organized into subfolders with a "table of contents" that would help everyone navigate to the information they seek more efficiently. Incidentally, i think this would also help with keyword search indexing.
